### PR TITLE
Feature/flower sizing and roots

### PR DESCRIPTION
**Pull request by kellydinneen**
## Is this a fix or a feature?
feature

## What is the change/fix?
- refactor petal creation to be dynamic
- size flowers according to repo lifespan
- create flower roots corresponding to repo branches

## Where should the reviewer start?
- Garden.js

### CODE REVIEW

File: src/components/Garden.js [hunk 1]
```diff
@@ -1,136 +1,162 @@
...
-
-      const animate = () => {
-        flowerBed.selectAll('.stem')
-          .transition().duration(2000)
-          .attr('d', d => `M0,600 C -20 ${yStemScale(d.lifespan)}, 20 ${yStemScale(d.lifespan)}, 0 ${yStemScale(d.lifespan)}`)
-            .attr('stroke', 'green')
-          .transition().duration(2000)
-          .attr('d', d => `M0,600 C 80 ${yStemScale(d.lifespan)}, -20 ${yStemScale(d.lifespan)}, 0 ${yStemScale(d.lifespan)}`)
-          .on("end", animate)
+      .range([gardenHeight - 300, gardenHeight - 350, gardenHeight - 400, gardenHeight - 450, gardenHeight/2, gardenHeight/3, gardenHeight/4])
```
Comment (CJSpohn) [inline src/components/Garden.js:22]: I think some of these are the same. GardenHeight - 400 is the same as gardenHeight/2 when it's set to be 800. This won't be the case if we change the height but right now I wonder what that's doing to our stems.

File: src/components/Garden.js [hunk 2]
```diff
@@ -1,136 +1,162 @@
...
+              .attr('stroke', 'green')
+            .transition().duration(2000)
+            .attr('d', d => `M0,600 C 80 ${yStemScale(d.lifespan)}, -20 ${yStemScale(d.lifespan)}, 0 ${yStemScale(d.lifespan)}`)
+            .on("end", animate)
+        }
+      animate();
+
+    const flowerSizeScale = d3.scaleQuantize()
+      .domain([minLifespan, maxLifespan])
+      .range([0.3, 0.35, 0.4, 0.45, 0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85])
```
Comment (CJSpohn) [inline src/components/Garden.js:48]: Nice!

File: src/components/Garden.js [hunk 3]
```diff
@@ -1,136 +1,162 @@
...
+            path: petalPaths[1],
+            petalRotationStart: petalRotationStarts[1],
+            scale: flowerSizeScale(d.lifespan)
+          },
+          {
+            color: colorsByLanguage[d.languages[0]] || 'none',
+            path: petalPaths[0],
+            petalRotationStart: petalRotationStarts[0],
+            scale: flowerSizeScale(d.lifespan)
+          },
```
Comment (CJSpohn) [inline src/components/Garden.js:96]: This is so smart. I never thought about giving it data based on an individual piece with a function

File: src/components/Garden.js [hunk 4]
```diff
@@ -1,136 +1,162 @@
...
+        .attr('fill', 'green')
+        .attr('stroke', 'green')
+        .attr('id', (d,i) => `${d.name + i}`)
+        .attr('d', () => {
+          const arrayOfLengths = [0, 110, 130, 150, 170, 180, 190, 210, 250];
+          const arrayOfCurves = [-20, -5, 5, 10, 15, 20, 25];
+          const index = (array) => [Math.floor(Math.random() * array.length)];
+          return `M0,0 C-10,40 ${arrayOfCurves[index(arrayOfCurves)]},100 0,${arrayOfLengths[index(arrayOfLengths)]}`;
+        })
+        .attr('transform', (d,i) => `rotate(${260 + (i + 1) * d.rotationFactor})`)
```
Comment (CJSpohn) [inline src/components/Garden.js:155]: So creative. Really, really cool. Why is the first element on the arrayOfLengths 0?

File: src/components/Garden.js [hunk 5]
```diff
@@ -1,136 +1,162 @@
...
+        .attr('xlink:href', (d, i) => `#myStem${i}`)
+        .text(d => d.name)
+        .attr('font-size', '1.5rem')
+
+      const rootBox = flowerBed.selectAll('.root-box')
+        .data(repositories).enter().append('svg')
+        .attr('class', '.root-box')
+        .attr('height', '200')
+        .attr('width', '200')
+        .attr('viewBox','-150 -150 300 300')
```
Comment (CJSpohn) [inline src/components/Garden.js:132]: Why is the viewbox starting at -150, -150?

File: src/components/Garden.js (hunk 1 above)
Comment (kellydinneen) [reply src/components/Garden.js:22]: oh yes, haha, you're right. However we do it, I just think we need to add more stem heights to the scale

File: src/components/Garden.js (hunk 4 above)
Comment (kellydinneen) [reply src/components/Garden.js:155]: I think that's a typo?

File: src/components/Garden.js (hunk 5 above)
Comment (kellydinneen) [reply src/components/Garden.js:132]: To make the center of the viewbox 0,0 and have the petals still fit without redoing their paths completely. In our original paths the petals were more than 100px long, so it was easiest to make the viewboxes 300x300

File: src/components/Garden.js (hunk 3 above)
Comment (kellydinneen) [reply src/components/Garden.js:96]: Yes, this really opened up D3 for me --- passing down and manipulating data from parent to child selections (much like you do from parent to child components in React)

File: src/components/Garden.js (hunk 4 above)
Comment (kellydinneen) [reply src/components/Garden.js:155]: I think we can consider a similar randomization of stem paths, but in that case we'll have to worry about the animation too

Comment (kellydinneen) [reply src/components/Garden.js:155]: another way of doing this that might be prettier: design 3+ full root paths that we store in an array and then randomly select entire path from that array

File: src/components/Garden.js (hunk 1 above)
Comment (kellydinneen) [reply src/components/Garden.js]: Latest commit fixes this

**Pull request merged**